service-base/src/main/java/org/eclipse/hono/service/credentials/BaseCredentialsService.java

```diff
@@ -80,47 +93,110 @@ protected String getEventBusAddress() {
         final String tenantId = request.getTenant();
         final JsonObject payload = request.getJsonPayload();
 
+        final Span span;
+        final Future<EventBusMessage> resultFuture;
         if (tenantId == null || payload == null) {
-            return Future.failedFuture(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST));
+            span = newChildSpan(SPAN_NAME_GET_CREDENTIALS, request.getSpanContext(), tenantId);
+            TracingHelper.logError(span, "missing tenant and/or payload");
+            resultFuture = Future.failedFuture(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST));
         } else {
             final String type = removeTypesafeValueForField(String.class, payload, CredentialsConstants.FIELD_TYPE);
             final String authId = removeTypesafeValueForField(String.class, payload,
                     CredentialsConstants.FIELD_AUTH_ID);
             final String deviceId = removeTypesafeValueForField(String.class, payload,
                     CredentialsConstants.FIELD_PAYLOAD_DEVICE_ID);
 
+            span = newChildSpan(SPAN_NAME_GET_CREDENTIALS, request.getSpanContext(), tenantId);
```

Comment:
there doesn't seem to be a difference between the *if* block and the *else* block regarding the creation of the span, is there?